### PR TITLE
feat(@deepnote/cli): add diff command to compare .deepnote files

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Add new `diff` command to compare structural differences between two .deepnote files:
- Shows added, removed, and modified notebooks
- Shows block-level changes within notebooks
- Includes summary statistics
- Supports --json output for scripting
- Supports --content flag to show content diffs

Includes comprehensive test coverage (27 tests).

## Summary by CodeRabbit

* **New Features**
  * Added a CLI "diff" command to compare two Deepnote files with JSON output, optional content-level diffs, and human-readable summary; shell completions updated.

* **Tests**
  * Added comprehensive test coverage for identical/different files, JSON/text outputs, content diffs, path handling, error cases, snapshots, renames, and edge cases.

* **Documentation**
  * CLI help and examples updated to include the diff command.

* **Chores**
  * Added diff fixtures and fixture-loading helpers.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: packages/cli/src/commands/diff.ts [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+import { decodeUtf8NoBom, deserializeDeepnoteFile } from '@deepnote/blocks'
+import chalk from 'chalk'
+import type { Command } from 'commander'
+import { diffLines } from 'diff'
+import { ExitCode } from '../exit-codes'
+import { debug, error as logError, output, outputJson } from '../output'
+import { FileResolutionError, resolvePathToDeepnoteFile } from '../utils/file-resolver'
+
+export interface DiffOptions {
+  output?: string
```
Comment (tkislan) [inline packages/cli/src/commands/diff.ts]: This can be a stricter type
```suggestion
  output?: 'json'
```

File: packages/cli/src/commands/diff.ts [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+  const file1 = deserializeDeepnoteFile(yamlContent1)
+
+  debug('Reading file 2...')
+  const rawBytes2 = await fs.readFile(absolutePath2)
+  const yamlContent2 = decodeUtf8NoBom(rawBytes2)
+  const file2 = deserializeDeepnoteFile(yamlContent2)
+
+  const diffResult = computeDiff(absolutePath1, absolutePath2, file1, file2, options)
+
+  if (options.output === 'json') {
```
Comment (tkislan) [inline packages/cli/src/commands/diff.ts]: If you change the type of `output` to literal, you can then change this to let typescript check that there are no more options to handle

File: packages/cli/src/commands/diff.ts (hunk 1 above)
Comment (dinohamzic) [reply packages/cli/src/commands/diff.ts]: Nice catch, fixed!

File: packages/cli/src/commands/diff.ts (hunk 2 above)
Comment (dinohamzic) [reply packages/cli/src/commands/diff.ts]: Done 👍
